src/hooks/useApi.ts

```diff
@@ -0,0 +1,35 @@
+import { useState, useCallback, useEffect } from "react";
+import { Obj } from "@/apis/apis.type";
+
+export function useApi<Params extends Obj, Response extends Obj>(
+  fetchFunc: (paramObj: Params) => Promise<Response>,
+  paramObj: Params
+) {
+  const [isLoading, setIsLoading] = useState<boolean>(true);
+  const [error, setError] = useState<Error | null>(null);
+  const [data, setData] = useState<Response | null>(null);
+
+  const wrappedFunc = useCallback(
+    async (paramObj: Params) => {
+      try {
+        setIsLoading(true);
+        setError(null);
+        const result = await fetchFunc(paramObj);
+        setData(result);
+      } catch (e) {
+        const error = e as Error;
+        setError(error);
+        console.error(error.message);
+      } finally {
```

Comment:
P2:
as 대신 아래와 같이 타입 체크하시는 것을 추천드립니다.
```suggestion
      } catch (error) {
        if(error instanceof Error) {
          setError(error);
          console.error(error.message);
        }
      } finally {
```